src/config.py

```diff
@@ -0,0 +1,222 @@
+allow.everyone.if.no.acl.found=false
+listener.name.sasl_plaintext.sasl.enabled.mechanisms=SCRAM-SHA-512
+"""
+
+
+class KafkaConfig:
+    """Manager for handling Kafka configuration."""
+
+    def __init__(self, charm: CharmBase):
+        self.charm = charm
+        self.default_config_path = SNAP_CONFIG_PATH
+        self.properties_filepath = f"{self.default_config_path}/server.properties"
+        self.jaas_filepath = f"{self.default_config_path}/kafka-jaas.cfg"
+
+    @property
+    def sync_password(self) -> Optional[str]:
+        """Returns charm-set sync_password for server-server auth between brokers."""
+        return self.charm.model.get_relation(PEER).data[self.charm.app].get("sync_password", None)
+
+    @property
+    def zookeeper_config(self) -> Dict[str, str]:
+        """The config from current ZooKeeper relations for data necessary for broker connection.
+
+        Returns:
+            Dict of ZooKeeeper `username`, `password`, `endpoints`, `chroot`, `connect` and `uris`
+        """
+        zookeeper_config = {}
+        for relation in self.charm.model.relations[ZK]:
+            zk_keys = ["username", "password", "endpoints", "chroot", "uris"]
+            missing_config = any(
+                relation.data[relation.app].get(key, None) is None for key in zk_keys
+            )
+
+            if missing_config:
+                continue
+
+            zookeeper_config.update(relation.data[relation.app])
+            break
+
+        if zookeeper_config:
+            sorted_uris = sorted(
+                zookeeper_config["uris"].replace(zookeeper_config["chroot"], "").split(",")
+            )
+            sorted_uris[-1] = sorted_uris[-1] + zookeeper_config["chroot"]
+            zookeeper_config["connect"] = ",".join(sorted_uris)
+
+        return zookeeper_config
+
+    @property
+    def zookeeper_connected(self) -> bool:
+        """Checks if there is an active ZooKeeper relation.
+
+        Returns:
+            True if ZooKeeper is currently related with sufficient relation data
+                for a broker to connect with. False otherwise.
+        """
+        if self.zookeeper_config.get("connect", None):
+            return True
+
+        return False
+
+    @property
+    def extra_args(self) -> List[str]:
+        """The necessary Java config options for SASL/SCRAM auth.
+
+        Returns:
+            List of Java config options
+        """
+        return [f"-Djava.security.auth.login.config={self.jaas_filepath}"]
+
+    @property
+    def bootstrap_server(self) -> List[str]:
+        """The current Kafka uris formatted for the `bootstrap-server` command flag.
+
+        Returns:
+            List of `bootstrap-server` servers
+        """
+        units: List[Unit] = list(
+            set([self.charm.unit] + list(self.charm.model.get_relation(PEER).units))
+        )
+        hosts = [
+            self.charm.model.get_relation(PEER).data[unit].get("private-address") for unit in units
+        ]
```

Comment:
Neat! This now works for getting the current unit's IP? Previously I had to get the current units IP differently like [this](https://github.com/canonical/mongodb-operator/blob/8796be02e10b84f990da0426ad1cff5922dc506a/src/charm.py#L411)

Reply:
I didn't know it didn't work! So far all my charms have been that way so I guess it works?